doc/release/upcoming_changes/14882.deprecation.rst

```diff
@@ -0,0 +1,30 @@
+Using the aliases of builtin types like ``np.int`` is deprecated
+----------------------------------------------------------------
+
+For a long time, ``np.int`` has been an alias of the builtin ``int``. This is
+repeatedly a cause of confusion for newcomers, and is also simply not useful.
+
+These aliases have been deprecated. The table below shows the full list of
+deprecated aliases, along with their exact meaning. Replacing uses of items in
+the first column with the contents of the second column will work identically
+and silence the deprecation warning.
+
+In many cases, it may have been intended to use the types from the third column.
+Be aware that use of these types may result in subtle but desirable behavior#
+changes.
+
+==================  =====================  ============================
+Deprecated          Equivalent to          Possibly intended numpy type
+==================  =====================  ============================
+``numpy.bool``      ``bool``               `numpy.bool_`
+``numpy.int``       ``int``                `numpy.int_` (default int dtype), `numpy.cint` (C ``int``)
+``numpy.float``     ``float``              `numpy.float_`, `numpy.double` (equivalent)
+``numpy.complex``   ``complex``            `numpy.complex_`, `numpy.cdouble` (equivalent)
+``numpy.object``    ``object``             `numpy.object_`
+``numpy.str``       ``str``                `numpy.str_`
+``numpy.long``      ``numpy.compat.long``  `numpy.int_` (C ``long``), `numpy.longlong` (largest integer type)
+``numpy.unicode``   ``numpy.compat.str``   `numpy.unicode_`
```

Comment:
```suggestion
``numpy.long``     ``int``                `numpy.int_` (C ``long``), `numpy.longlong` (largest integer type)
``numpy.unicode``   ``str``                `numpy.unicode_`
```

Reply:
Just because we've dropped py2 doesn't mean that people using numpy aren't supporting both python2 and python 3.

Reply:
Ah, good point, did not think of that, hmmmm. Can we write "``int`` (``long`` on Python 2)" maybe?  The `compat` stuff seems confusing to me...

Reply:
Rethinking about it, maybe we should propagate use `float64` and `float32` instead of `float` and `double`?

Reply:
Yes, that phr

> Can we write "`int` (`long` on Python 2)" maybe?

Sure, that would be fine. The `numpy.compat.str` entry should read `numpy.compat.unicode` anyway.

Reply:
> `float64` and `float32` instead of `float` and `double`?

Not sure what you mean here - none of these rows are about `float32`, they're only regarding the 64-bit floats.

Reply:
Oh, the `Possibly intended numpy type` column: ``` `numpy.float_`, `numpy.double` (equivalent)```. You are right, I thought that `np.float_` might be a 32bit one for a moment there...  Anyway, we can keep it as is, not sure that `numpy.float64` would be much nicer than `numpy.float_`.

In any case, this would matter for the DeprecationWarning much more than for this table.